### PR TITLE
nixos/soteria: init module

### PR DESCRIPTION
**Pull request by JohnRTitor**
This adds a `security.soteria.enable` option.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
- [x] Tested, as applicable:
  - Tested with my own configuration
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/security/soteria.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  options.security.soteria = {
+    enable = lib.mkEnableOption null // {
+      description = ''
+        Whether to enable Soteria,
+          Polkit authentication agent for any desktop environment.
+
+          ::: {.note}
+          You should only enable this if you are on a Desktop Environment that
+          does not provide a graphical polkit authentication agent, or you are on
+          a standalone window manager like Hyprland and Sway.
```
Comment (fufexan) [inline nixos/modules/security/soteria.nix:21]:
```suggestion
          a standalone window manager or Wayland compositor.
```

File: nixos/modules/security/soteria.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+let
+  cfg = config.security.soteria;
+in
+{
+  options.security.soteria = {
+    enable = lib.mkEnableOption null // {
+      description = ''
+        Whether to enable Soteria,
+          Polkit authentication agent for any desktop environment.
```
Comment (fufexan) [inline nixos/modules/security/soteria.nix:16]: Weird formatting here.

**Review by JohnRTitor — COMMENTED** [review]
Should be ready now. I wonder if I should name it `polkit-soteria`, so that people can discover it easily.

File: nixos/modules/security/soteria.nix (hunk 2 above)
Comment (JohnRTitor) [reply nixos/modules/security/soteria.nix:16]: Nixfmt-rfc-style did that 🤦

File: nixos/modules/security/soteria.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  options.security.soteria = {
+    enable = lib.mkEnableOption null // {
+      description = ''
+        Whether to enable Soteria, a Polkit authentication agent
+        for any desktop environment.
+
+        ::: {.note}
+        You should only enable this if you are on a Desktop Environment that
+        does not provide a graphical polkit authentication agent, or you are on
+        a standalone window manager like or Wayland compositor.
```
Comment (fufexan) [inline nixos/modules/security/soteria.nix:21]:
```suggestion
        a standalone window manager or Wayland compositor.
```

**Comment by NotAShelf** [comment]
Currently unable to test, but diff LGTM.

**Review by NotAShelf — APPROVED** [review]

**Review by fufexan — APPROVED** [review]

**Pull request merged**